config/config.go

```diff
@@ -946,6 +947,7 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetDefault("adapters.silvermob.endpoint", "http://{{.Host}}.silvermob.com/marketplace/api/dsp/bid/{{.ZoneID}}")
 	v.SetDefault("adapters.smaato.endpoint", "https://prebid.ad.smaato.net/oapi/prebid")
 	v.SetDefault("adapters.smartadserver.endpoint", "https://ssb-global.smartadserver.com")
+	v.SetDefault("adapters.smarthub.endpoint", "http://{{.Host}}/?seat={{.AccountID}}&token={{.SourceId}}")
```

Comment:
Unfortunately, impossible for us. Replacing of all host is the only one way how we work with our partners. We need to leave this how it is. BTW, there are a lot of adapters with the same pattern. Several examples:

```
858	v.SetDefault("adapters.adkernel.endpoint", "http://{{.Host}}/hb?zone={{.ZoneID}}")
config/config.go
```
[link](https://github.com/prebid/prebid-server/blob/bab487e8a54b37dca75c5f86694170e60d638ef3/config/config.go#L858)
```
918	v.SetDefault("adapters.invibes.endpoint", "https://{{.Host}}/bid/ServerBidAdContent")
config/config.go
```
[link](https://github.com/prebid/prebid-server/blob/bab487e8a54b37dca75c5f86694170e60d638ef3/config/config.go#L918)
```
978	v.SetDefault("adapters.zeroclickfraud.endpoint", "http://{{.Host}}/openrtb2?sid={{.SourceId}}")
config/config.go
```
[link](https://github.com/prebid/prebid-server/blob/bab487e8a54b37dca75c5f86694170e60d638ef3/config/config.go#L978)

Reply:
@SmartHubSolutions We're aware that there are a few other adapters that use this approach for the hosts but it's an anti-pattern that we as a Prebid Server committee strongly oppose and are currently working with these adapters to correct their behavior. We're focused on making sure that this pattern isn't used for any other adapters going further. It's odd to ask the publisher to specify the server address of the bidder, it greatly harms Prebid Server's performance which is based on keep alive connections to bidders, and allowing the publisher to pass the host through opens up PBS to a variety of attacks.

Can you please help us understand your use case for this? We'd be happy to work with you to find a better solution for your use case, if possible. 

In case the use case here is to have the Publishers specify the host based on the region they are in then the recommended approach for that is to use a default endpoint pointing to your company's global load balancer. If you don't have a global load balancer, consider marking the adapter as disabled by default and add instructions in your bidder docs to let hosts know which endpoints to use for their geo regions, or ask them to contact you directly to setup if you don't want to publicly post those endpoints.

Reply:
@mansinahar Thank you for this detailed explanation! Now, everything is clear. I'll discuss this with my team to find a solution. We'll definitely find it and back soon.

Reply:
@mansinahar I've written below how we had changed URL template.Is it meet requirements now? :) Can we continue our review process?